Mathlib/Data/IsROrC/Basic.lean

```diff
@@ -822,8 +824,27 @@ noncomputable instance Real.isROrC : IsROrC ℝ where
   norm_sq_eq_def_ax z := by simp only [sq, Real.norm_eq_abs, ← abs_mul, abs_mul_self z, add_zero,
     mul_zero, AddMonoidHom.zero_apply, AddMonoidHom.id_apply]
   mul_im_I_ax z := by simp only [MulZeroClass.mul_zero, AddMonoidHom.zero_apply]
+  le_iff_re_im := by simp only [AddMonoidHom.id_apply, AddMonoidHom.zero_apply, and_true,
+    forall_const]
 #align real.is_R_or_C Real.isROrC
 
+instance toStarOrderedRingK : StarOrderedRing K := by
+  apply StarOrderedRing.ofNonnegIff'
+  · intros x y hxy z
+    rw [IsROrC.le_iff_re_im] at *
+    simpa [map_add, add_le_add_iff_left, add_right_inj] using hxy
+  · intro x
+    rw [IsROrC.le_iff_re_im, map_zero, map_zero, IsROrC.star_def, eq_comm]
+    constructor
+    · intro hx
+      use (Real.sqrt (IsROrC.re x))
+      rw [IsROrC.conj_ofReal, ← IsROrC.ofReal_mul, Real.mul_self_sqrt hx.1, IsROrC.ext_iff,
+        IsROrC.ofReal_re, IsROrC.ofReal_im, eq_self, true_and, hx.2]
+    · simp only [IsROrC.star_def, IsROrC.conj_mul, forall_exists_index]
+      intro y hy
+      rw [hy, IsROrC.ofReal_re, IsROrC.ofReal_im, eq_self, and_true]
+      apply IsROrC.normSq_nonneg
```

Comment:
Did you prove this from scratch? Can you instead try to adapt the old proof which was
```lean
  StarOrderedRing.ofNonnegIff' add_le_add_left fun r => by
    refine' ⟨fun hr => ⟨Real.sqrt r.re, _⟩, fun h => _⟩
    · have h₁ : 0 ≤ r.re := by
        rw [le_def] at hr
        exact hr.1
      have h₂ : r.im = 0 := by
        rw [le_def] at hr
        exact hr.2.symm
      ext
      · simp only [ofReal_im, star_def, ofReal_re, sub_zero, conj_re, mul_re, mul_zero,
          ← Real.sqrt_mul h₁ r.re, Real.sqrt_mul_self h₁]
      · simp only [h₂, add_zero, ofReal_im, star_def, zero_mul, conj_im, mul_im, mul_zero,
          neg_zero]
    · obtain ⟨s, rfl⟩ := h
      simp only [← normSq_eq_conj_mul_self, normSq_nonneg, zero_le_real, star_def]
```

Reply:
Ah nevermind, I see the proof is hard to replicate.